src/session_impl.cpp

```diff
@@ -4675,14 +4675,10 @@ namespace aux {
 		torrent_ptr->start(params);
 
 #ifndef TORRENT_DISABLE_EXTENSIONS
-		typedef std::vector<boost::function<
-			boost::shared_ptr<torrent_plugin>(torrent_handle const&, void*)> >
-			torrent_plugins_t;
-
-		for (torrent_plugins_t::const_iterator i = params.extensions.begin()
-			, end(params.extensions.end()); i != end; ++i)
+		for (auto& ext : params.extensions)
 		{
-			torrent_ptr->add_extension((*i)(handle, params.userdata));
+			boost::shared_ptr<torrent_plugin> tp(ext(handle, params.userdata));
+			if (tp) torrent_ptr->add_extension(tp);
```

Comment:
ok, I will change it here and in another place. I'm not sure what's is more expensive in this case btw.